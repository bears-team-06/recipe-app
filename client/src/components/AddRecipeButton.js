import React from "react";
import styled from 'styled-components'

const Button = styled.button`
  font-size: 40px;
  color: ${props => props.theme.textColor};
  background: ${props => props.theme.secondary};
  border: 0;
  width: 45px;
  height: 45px;
  border-radius: 50%;
  border: 0px solid rgba(255, 255, 255, 0);
  box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23); 
  &:hover {
    background: ${props => props.theme.secondaryLight};
  }
`

const AddRecipeButton = (props) => {
    return <Button onClick={props.onClick}>+</Button>
}


export default AddRecipeButton