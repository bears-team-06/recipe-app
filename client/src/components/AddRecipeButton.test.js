import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import AddRecipeButton from "./AddRecipeButton";

describe("Add Recipe Button", () => {
  it("matches snapshot", () => {
    const RecipeButtonWrapper = renderer
      .create(<AddRecipeButton onClick={() => {}} />)
      .toJSON();
    expect(RecipeButtonWrapper).toMatchSnapshot();
  });

  it("calls prop onClick when clicked", () => {
    const onClickSpy = jest.fn();
    const RecipeButtonWrapper = mount(<AddRecipeButton onClick={onClickSpy} />);
    RecipeButtonWrapper.find("svg").simulate("click");
    expect(onClickSpy).toHaveBeenCalledTimes(1);
  });
});
