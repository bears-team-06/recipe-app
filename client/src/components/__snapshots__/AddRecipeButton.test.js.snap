// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Recipe Button matches snapshot 1`] = `
.c0 {
  font-size: 40px;
  border: 0;
  width: 45px;
  height: 45px;
  border-radius: 50%;
  border: 0px solid rgba(255,255,255,0);
  box-shadow: 0 10px 20px rgba(0,0,0,0.19),0 6px 6px rgba(0,0,0,0.23);
}

<button
  className="c0"
  onClick={[Function]}
>
  +
</button>
`;
